File: Data Structures & Algorithms/max-water-container/submission-15.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        int left=0;
        int right=nums.size()-1;
        int ans=0;
        while(left<=right)
        {
            if(nums[left] < nums[right])
            {
                ans=max(ans,nums[left]*(right-left));
                left++;
            }
            else{
                ans=max(ans,nums[right]*(right-left));
                right--;
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/sort-colors/submission-15.java
class Solution {
    public void sortColors(int[] nums) {
        int l=0;
        int m=0;
        int r=nums.length-1;
        while(m<=r)
        {
            if(nums[m]==0)
            {
                int x=nums[l];
                nums[l]=nums[m];
                nums[m]=x;
                m++;
                l++;
            }
            else if(nums[m]==1)
            {
                m++;
            }
            else{
                int y=nums[m];
                nums[m]=nums[r];
                nums[r]=y;
                r--;
            }
        }
    }
}